src/Invio.Extensions.Linq.Async/IQueryableExtensions.cs

```diff
@@ -0,0 +1,1469 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Linq.Expressions;
+using System.Reflection;
+using System.Threading.Tasks;
+using Invio.Extensions.Reflection;
+
+namespace Invio.Extensions.Linq.Async {
+    /// <summary>
+    /// Extension methods for Asynchronous operations on <see cref="IQueryable{T}" />.
+    /// </summary>
+    public static class IQueryableExtensions {
+        /// <summary>
+        /// Asynchronous version of <see cref="Enumerable.ToList{T}" />.
+        /// </summary>
+        public static async Task<List<T>> ToListAsync<T>(this IQueryable<T> source) {
```

Comment:
More specifically, I would expect `ToImmutableListAsync()`, and `ToImmutableHashSetAsync()`, and `ToImmutableDictionaryAsync()` implementations to make an appearance in this repository since those are asynchronous implementations of what I see us use the most often in our closed source code.